src/codelabs/iterables.md

```diff
@@ -0,0 +1,608 @@
+
+<?code-excerpt "iterables/test/iterables_test.dart (iterable-elementat)"?>
+{% prettify dart %}
+Iterable<int> iterable = [1, 2, 3];
+int value = iterable.elementAt(1);
+{% endprettify %}
+
+Continue to the next section to learn more about how to access elements of an `Iterable`.
+
+## Reading elements
+
+You can read the elements of an iterable sequentially, using a `for-in` loop.
+
+### Example: Using a for-in loop
+
+The following example shows you how to read elements using  a `for-in` loop.
+
+[//]: https://gist.github.com/c419f595f95e8317c54192491ae017cd
+<iframe
+  src="{{site.dartpad-embed}}?id=c419f595f95e8317c54192491ae017cd"
+  style="border: 1px solid lightgrey; margin-top: 10px; margin-bottom: 25px"
+  frameborder="no"
+  height="200"
+  width="100%" >
+</iframe>
+
+{{site.alert.info}}
+Behind the scenes, the `for-in` loop uses an [iterator.]({{site.dart_api}}/stable/dart-core/Iterable/iterator.html)
+You rarely see the [Iterator API][iterator class] used directly, however,
+because `for-in` is easier to read and understand,
+and is less prone to errors.
+{{site.alert.end}}
+
+{{site.alert.secondary}}
+**Key terms:**
+* **Iterable**: The Dart [Iterable][iterable class] class.
+* **Iterator**: An object used by `for-in` to read elements from an `Iterable` object.
+* **`for-in` loop**: An easy way to sequentially read elements from an `Iterable`.
+{{site.alert.end}}
+
+### Example: Using first and last
+
+In some cases, you only want to access the first or the last element of an `Iterable`.
+
+With the `Iterable` class, you can't access the elements directly, so you can't call
+`iterable[0]` to access the first element. Instead, you can use `first`,
+which gets the first element.
+
+Also, with the `Iterable` class, you can't use the operator `[]` to access the last element, but you can use the
+`last` property. 
+
+{{site.alert.warn}}
+  Because accessing the last element of an Iterable requires stepping through all the other elements,
+  **`last` can be slow.**
+  Using `first` or `last` on an **empty `Iterable`**
+  results in a [StateError.][StateError class]
+{{site.alert.end}}
+
+
+[//]: https://gist.github.com/f3d96039195566d934a966390ea4ad62
+<iframe
+  src="{{site.dartpad-embed}}?id=f3d96039195566d934a966390ea4ad62"
+  style="border: 1px solid lightgrey; margin-top: 10px; margin-bottom: 25px"
+  frameborder="no"
+  height="300"
+  width="100%" >
+</iframe>
+
+In this example, you saw how to use `first` and `last` to get the first
+and last elements of an `Iterable`. It's also possible to
+find the first element that matches a condition. The next section
+shows how to do that using a method called `firstWhere()`.
+
+### Example: Using firstWhere()
+
+You already saw that you can access the elements of an `Iterable` sequentially,
+and you can easily get the first or last element.
+
+Now, you learn how to use `firstWhere()` to find an element that
+satisfies certain conditions. This method requires you to pass a _predicate_,
```

Comment:
```suggestion
Now, you learn how to use `firstWhere()` to find the first element in the 
`Iterable` that satisfies certain conditions. This method requires you to pass a _predicate_,
```

Consider adding a subsequent element in `items` that satisfies the condition (to demonstrate that `firstWhere()` won't return this element)